src/server/utils/requester/requester.ts

```diff
@@ -17,40 +17,45 @@
 
 import { concurrentLimitRequesterFactory, retryRequesterFactory, verboseRequesterFactory } from "plywood";
 import { PlywoodRequester } from "plywood-base-api";
-import { DruidRequestDecorator, druidRequesterFactory, Protocol } from "plywood-druid-requester";
-import { SupportedType } from "../../../common/models/cluster/cluster";
+import { DruidRequestDecorator, druidRequesterFactory } from "plywood-druid-requester";
+import { Cluster } from "../../../common/models/cluster/cluster";
 import { threadConditionally } from "../../../common/utils/functional/functional";
 
 export interface ProperRequesterOptions {
-  type: SupportedType;
-  host: string;
+  cluster: Cluster;
   retry?: number;
-  timeout?: number;
   verbose?: boolean;
   concurrentLimit?: number;
-
-  // Specific to type 'druid'
   druidRequestDecorator?: DruidRequestDecorator;
-  protocol?: Protocol;
+}
 
-  // Specific to SQL drivers
-  database?: string;
-  user?: string;
-  password?: string;
+function protocolTranslator(protocol: string): "plain" | "tls" {
+  if (protocol === "https:") return "tls";
+  return "plain";
 }
 
-function createRequester({ type, host, timeout, druidRequestDecorator, protocol }: ProperRequesterOptions): PlywoodRequester<any> {
-  switch (type) {
-    case "druid":
-      return druidRequesterFactory({
-        host,
-        timeout,
-        requestDecorator: druidRequestDecorator,
-        protocol
-      });
-    default:
-      throw new Error(`unknown requester type ${type}`);
+function defaultPorts(url: URL): URL {
+  if (url.port) return url;
+  if (url.protocol === "http:") {
+    url.port = "80";
+    return url;
+  }
+  if (url.protocol === "https:") {
+    url.port = "443";
+    return url;
   }
+  return url;
+}
+
+function createDruidRequester(cluster: Cluster, druidRequestDecorator: DruidRequestDecorator): PlywoodRequester<any> {
+  const { href, protocol } = defaultPorts(new URL(cluster.url));
```

Comment:
`defaultPorts` is used deep down there, because only `druidRequestfactory` has weird logic about default ports. We don't want to pollute url with forced port before this call